File: src/main/kotlin/inflearn/java/ch01_string/_1_8_유효한_팰린드롬.java
package inflearn.java.ch01_string;

import java.util.Scanner;

public class _1_8_유효한_팰린드롬 {

    public static String solution(String str) {
        StringBuilder sb = new StringBuilder();
        for(int i = 0; i < str.length(); i++) {
            if(Character.isAlphabetic(str.charAt(i)))
                sb.append(str.charAt(i));
        }
        String origin = sb.toString();
        String reverse = sb.reverse().toString();

        if(origin.equalsIgnoreCase(reverse)) return "YES";
        else return "NO";
    }

    /**
     * 정규식을 활용하는 방법
     * ^ -> 아닌 것들
     * [^A-Z] -> A부터 Z 까지가 아닌 것들
     * @param s
     * @return
     */
    public static String solution2(String s) {
        s = s.toUpperCase().replaceAll("[^A-Z]","");
        String tmp = new StringBuilder(s).reverse().toString();
        if(s.equals(tmp))
            return "YES";
        return "NO";
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();

        System.out.println(solution(input));
    }
}

File: src/main/kotlin/inflearn/1_8 유효한 팰린드롬.kt
package inflearn

import inflearn.java._1_8_유효한_팰린드롬
import java.util.*

object `1_8 유효한 팰린드롬` {
    fun solution(str: String): String {
        val sb = StringBuilder(str)
        sb.reverse()
        return if (sb.toString().equals(str, ignoreCase = true)) "YES" else "NO"
    }

    /**
     * 정규식을 활용하는 방법
     * ^ -> 아닌 것들
     * [^A-Z] -> A부터 Z 까지가 아닌 것들
     * @param s
     * @return
     */
    fun solution2(s: String): String {
        val str = s.uppercase().replace(regex = "[^A-Z]".toRegex(), replacement = "")
        val tmp = StringBuilder(str).reverse().toString()
        return if (str == tmp) "YES" else "NO"
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val sc = Scanner(System.`in`)
        val input = sc.nextLine()
        println(solution(input))
    }
}